packages/create-expo-module/src/create-expo-module.ts

```diff
@@ -0,0 +1,271 @@
+  try {
+    const { stdout } = await spawnAsync('npm', ['whoami'], { cwd: targetDir });
+    return stdout.trim();
+  } catch (e) {
+    return null;
+  }
+}
+
+/**
+ * Downloads the template from NPM registry.
+ */
+async function downloadPackageAsync(targetDir: string): Promise<string> {
+  const tarballUrl = await getNpmTarballUrl('expo-module-template');
+
+  await downloadTarball({
+    url: tarballUrl,
+    dir: targetDir,
+  });
+  return path.join(targetDir, 'package');
+}
+
+/**
+ * Asks the user for some data necessary to render the template.
+ * Some values may already be provided by command options, the prompt is skipped in that case.
+ */
+async function askForSubstitutionDataAsync(
+  targetDir: string,
+  options: CommandOptions
+): Promise<SubstitutionData> {
+  const defaultPackageSlug = path.basename(targetDir);
+  const defaultProjectName = defaultPackageSlug
+    .replace(/^./, (match) => match.toUpperCase())
+    .replace(/\W+(\w)/g, (_, p1) => p1.toUpperCase());
+
+  const { slug } = options.target
+    ? { slug: defaultPackageSlug }
+    : await prompts({
+        type: 'text',
+        name: 'slug',
+        message: 'What is the package slug?',
+        initial: defaultPackageSlug,
+      });
+
+  const { name } = options.name
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'name',
+        message: 'What is the project name?',
+        initial: defaultProjectName,
+      });
+
+  const { description } = options.description
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'description',
+        message: 'How would you describe the module?',
+      });
+
+  const { package: projectPackage } = options.package
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'package',
+        message: 'What is the Android package name?',
+        initial: `expo.modules.${defaultPackageSlug.replace(/\W/g, '').toLowerCase()}`,
+      });
+
+  const { author } = options.author
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'author',
+        message: 'Who is the author?',
+        initial: (await npmWhoamiAsync(targetDir)) ?? '',
+      });
+
+  const { license } = options.license
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'license',
+        message: 'What is the license?',
+        initial: 'MIT',
+      });
+
+  const { repo } = options.repo
+    ? options
+    : await prompts({
+        type: 'text',
+        name: 'repo',
+        message: 'What is the repository URL?',
+      });
+
+  return {
+    project: {
+      slug,
+      name,
+      version: '0.1.0',
+      description,
+      package: projectPackage,
+    },
+    author,
+    license,
+    repo,
+  };
+}
+
+const program = new Command();
+
+program
+  .name(packageJson.name)
+  .version(packageJson.version)
+  .description(packageJson.description)
+  .arguments('[target_dir]')
```

Comment:
Is there `.usage` part available?
Description taken from `package.json` is fine, but we should have a simple example visible instantly too 😉